File: Source Code/UnmanagedDemo/Source/PushButton.cpp
#include "stdafx.h"
#include "PushButton.h"
#include "Demo.h"

PushButton::PushButton(DemoRectangle rect, Texture2D* tex, DemoColour col) : DemoButton(rect, tex, col)
{
	isPressed = false;
}


PushButton::~PushButton(void)
{
}

void PushButton::Depress()
{
	isPressed = true;
	demo->SpawnObject();
}

void PushButton::Unpress()
{
	isPressed = false;
}

File: Source Code/.Net Demo/Source/PushButton.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;

namespace DotNetDemo
{
    public class PushButton : DemoButton
    {
        public PushButton(Rectangle rect, Color col)
            : base(rect, col)
        {
            isPressed = false;
        }

        public override void Depress()
        {
            isPressed = true;
            demo.SpawnObject();
        }

        public override void Unpress()
        {
            isPressed = false;
        }
    }
}
